firmware/user/modes/mode_ddr.c

```diff
@@ -580,7 +638,40 @@ static void ICACHE_FLASH_ATTR ddrHandleArrows(void)
 static void ICACHE_FLASH_ATTR ddrAnimateNotes(void* arg __attribute__((unused)))
 {
     ddr->NoteIdx = (ddr->NoteIdx + 1) % 4;
-    // testUpdateDisplay();
+}
+
+static void ICACHE_FLASH_ATTR ddrMakeScoreString(ddrWinResult_t curWin, char* scoreText, unsigned int size)
```

Comment:
Have you thought about checking for a nonzero score when building the string? Having a bunch of zero scores with 'C' ratings the first time you check the scores is a little odd. I'd either not show a letter grade, or return an empty string.